src/Components/Components/src/NavigationManager.cs

```diff
@@ -201,7 +201,14 @@ internal static string NormalizeBaseUri(string baseUri)
         /// </summary>
         protected void NotifyLocationChanged(bool isInterceptedLink)
         {
-            _locationChanged?.Invoke(this, new LocationChangedEventArgs(_uri, isInterceptedLink));
+            try
+            {
+                _locationChanged?.Invoke(this, new LocationChangedEventArgs(_uri, isInterceptedLink));
+            }
+            catch (Exception ex)
+            {
+                throw new LocationChangeException("An exception occurred while dispatching an location changed event.", ex);
```

Comment:
nit: I think it should be "a" not "an"
```suggestion
                throw new LocationChangeException("An exception occurred while dispatching a location changed event.", ex);
```

Reply:
I could change it to say that it's `an historic location changed event`

Reply:
I think it is still "a" in that case.

Reply:
I took your sugestion but that doesn't stop me from trolling. I might also change it to:

```
OOF. An exception whom'st've been occurred while dispatching an historic location changed event.
```